test/stream-monitor/sm_plugin/stream_monitor.py

```diff
@@ -45,7 +51,7 @@ def configure(self, options, conf):
         super(StreamMonitorPlugin, self).configure(options,conf)
         if not self.enabled:
             return
-        if conf.options.collect_only:
+        if getattr(conf.options, 'collect_only', False):
```

Comment:
You won't see logging from within the tests as they are probed. 
Note that this exact change (using getattr) leaves the behavior the same for "normal" tests, but allows the plugin self-tests to be run. 

Would love to totally kill all logging (and log directory rotation) for -list, but that is a much much (MUCH!) trickier thing to solve.